pkg/hubble/relay/server/option.go

```diff
@@ -165,3 +161,13 @@ func WithInsecureClient() Option {
 		return nil
 	}
 }
+
+// WithClusterName sets the cluster name for the peer service
+// so that it knows how to construct the proper TLSServerName
+// to validate mTLS in the K8s Peer service.
+func WithClusterName(clusterName string) Option {
```

Comment:
Trying to think about the clustermesh case where there are more than one cluster: Relay will connect to the peer svc on the same cluster. Would `WithLocalClusterName` make sense here?